zipkin-storage/cassandra3/src/main/java/zipkin/storage/cassandra3/CassandraUtil.java

```diff
@@ -73,19 +73,20 @@ public static int durationIndexBucket(long ts_micro) {
       if (Constants.CORE_ANNOTATIONS.contains(a.value)) continue;
 
       if (a.endpoint != null && !a.endpoint.serviceName.isEmpty()) {
-        annotationKeys.add(a.endpoint.serviceName + ":" + a.value);
+        annotationKeys.add(a.endpoint.serviceName);
+        annotationKeys.add(a.value);
       }
     }
     for (BinaryAnnotation b : span.binaryAnnotations) {
-      if (b.type == BinaryAnnotation.Type.STRING
-          && b.endpoint != null
-          && !b.endpoint.serviceName.isEmpty()
-          && b.value.length <= LONGEST_VALUE_TO_INDEX * 4) { // UTF_8 is up to 4bytes/char
-        String value = new String(b.value, UTF_8);
-        if (value.length() > LONGEST_VALUE_TO_INDEX) continue;
-
-        annotationKeys.add(b.endpoint.serviceName + ":" + b.key);
-        annotationKeys.add(b.endpoint.serviceName + ":" + b.key + ":" + new String(b.value, UTF_8));
+      if (b.endpoint != null && !b.endpoint.serviceName.isEmpty() && !Constants.CORE_ANNOTATIONS.contains(b.key)) {
+        annotationKeys.add(b.endpoint.serviceName);
+        if (b.type == BinaryAnnotation.Type.STRING) {
+            String value = new String(b.value, UTF_8);
+            value = value.substring(0, Math.min(value.length(), LONGEST_VALUE_TO_INDEX));
+            annotationKeys.add(b.key + ":" + value);
+        } else {
+            annotationKeys.add(b.key);
```

Comment:
>
> feasibility of it aside, has it value?
>
> No, this is value-reducing. For example, we would have to increase scope
of other things to make sure special-case boolean annotations like "sa"
aren't accidentally now indexed.

Moreover, there are numerous problems in the subtleties of thrift types.
When we start indexing them by default, we are reversing our position (that
discouraged them), and would have to take responsibility to support them
despite their problems. For example, this makes transitioning to a new
model that doesn't have an I16 type more difficult. It raises questions on
lossiness of I64 when rendered in json (happened last week!) Right now, the
api is pretty clear about text annotations only, and this is what we should
be doing.